company-ycmd.el

```diff
@@ -213,18 +219,35 @@ string as text-properties, and returns the string."
      ((string-match "\\((.*)[ a-z]*\\'\\)" params)
       (match-string 1 params)))))
 
+(defun company-ycmd--get-kind (candidate)
+  "Get information about completion kind from CANDIDATE."
+  (-when-let (kind (get-text-property 0 'kind candidate))
+    (pcase kind
+      ("STRUCT" "struct")
+      ("CLASS" "class")
+      ("ENUM" "enum")
+      ("TYPE" "type")
+      ("MEMBER" "member")
+      ("FUNCTION" "fn")
+      ("VARIABLE" "var")
+      ("MACRO" "macro")
+      ("PARAMETER" "parameter")
+      ("NAMESPACE" "namespace"))))
+
+(defun company-ycmd--get-return-type (candidate)
+  "Get return type of CANDIDATE."
+  (s-present?
+   (or (get-text-property 0 'return_type candidate)
+       (get-text-property 0 'extra_menu_info candidate))))
+
 (defun company-ycmd--annotation (candidate)
-  "Fetch the annotation text-property from a candidate string."
-  (let ((kind (get-text-property 0 'kind candidate))
-        (return-type (or (get-text-property
-                          0 'return_type candidate)
-                         (get-text-property
-                          0 'extra_menu_info candidate))))
+  "Fetch the annotation text-property from a CANDIDATE string."
+  (let ((kind (and company-ycmd-show-completion-kind
+                   (company-ycmd--get-kind candidate)))
+        (return-type (company-ycmd--get-return-type candidate)))
```

Comment:
I merged this without testing it, but there seems to be a problem here. `company-ycmd--get-return-type` returns a boolean, i.e. the result of `s-present?`. But later we seems to treat that return value as a string in the call to `concat`. 

Is this a bug in `company-ycmd--get-return-type` or in our the use of that function here? Or am I misunderstanding things?